### PR TITLE
Modal Component

### PR DESCRIPTION
**Pull request by rs6g10**
# Modal Component

Adding a react modal component that uses bootstrap styles and animation.
It handles the following:
- Escape to close modal
- Modal with standard sizes.
- Fullscreen on mobile. This work is discussed with Mike Marter (@pixeljunkies ) who has confirmed the design for modals going full screen on mobile. It is recommended to use only one button in Mobile footer since it takes full width on mobile. If you need multiple buttons, you can stack them on each other, take guidance from your fellow designer. The styles will be moved to Transferwise bootstrap styles so we include no CSS in this project for modal.

### How does it look like?
modal-desktop
modal-mobile

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+## Add modal component
+
+![modal-mobile](https://user-images.githubusercontent.com/1889234/44737897-20da1600-aaeb-11e8-8f76-288598d30b41.gif)
+![modal-desktop](https://user-images.githubusercontent.com/1889234/44737899-20da1600-aaeb-11e8-8bc8-ac993273423c.gif)
+
 # v4.3.5
 ## Parse MoneyInput inputs as positive numbers
 
-Since you're entering money, you can't enter negative numbers. When you enter a negative amount now, it'll be parsed as positive.
+Since you're eering money, you can't enter negative numbers. When you enter a negative amount now, it'll be parsed as positive.
```
Comment (Tankenstein) [inline CHANGELOG.md]: What happened here 😄

File: docs/ModalDocs.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      </button>
+    );
+
+    return (
+      <div className="container">
+        <section className="section">
+          <div className="row">
+            <div className="col-md-6">
+              <h2>Modal</h2>
+              <p>That works on mobile too!</p>
```
Comment (Tankenstein) [inline docs/ModalDocs.js]: For some reason, we usually put some dumb tagline here (America's Next Top Modal), but feel free to ignore this one 😄

File: CHANGELOG.md (hunk 1 above)
Comment (rs6g10) [reply CHANGELOG.md]: D'oh!

File: docs/ModalDocs.js (hunk 2 above)
Comment (rs6g10) [reply docs/ModalDocs.js]: I'll add America's Next Top Modal 😂

File: src/modal/ModalDialog/ModalDialog.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import { modalOpenOperations, modalCloseOperations } from './DOMOperations';
+import './ModalDialog.less';
+
+export default class ModalDialog extends Component {
+  static propTypes = dialogPropTypes;
+  static defaultProps = dialogDefaultProps;
+
+  constructor(props) {
+    super(props);
+    this.onEscape = this.onEscape.bind(this);
```
[the remark this answers is not in the document]
Comment (rs6g10) [reply src/modal/ModalDialog/ModalDialog.js]: I'll change it. Though I personally like binding 😹

File: src/modal/props.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import Types from 'prop-types';
```
Comment (Tankenstein) [inline src/modal/props.js]: I would keep this with the modal file itself, that way you can immediately discern what the props are for a component :)

Comment (Tankenstein) [reply src/modal/props.js]: Ah thanks, I didn't realise, was skimming.

File: src/modal/ModalDialog/DOMOperations/DOMOperations.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+export function modalOpenOperations() {
+  addModalOpenBodyClass();
+}
+
+export function modalCloseOperations() {
+  removeModalOpenBodyClass();
+}
+
+function addModalOpenBodyClass() {
```
Comment (Tankenstein) [inline src/modal/ModalDialog/DOMOperations/DOMOperations.js]: I think these two functions are unnecessary, pretty self-explanatory, not really adding abstraction on top of it, wdyt?

Comment (oliverviljamaa) [reply src/modal/ModalDialog/DOMOperations/DOMOperations.js]: They were created with the assumption that there will be more operations. I would actually remove the `modalOpenOperations` and `modalCloseOperations` and export `addModalOpenBodyClass` and `removeModalOpenBodyClass` straight. WDYT?

Comment (rs6g10) [reply src/modal/ModalDialog/DOMOperations/DOMOperations.js]: Thanks Gents! I'll change it.

**Pull request merged**